torchmetrics/functional/audio/pesq.py

```diff
@@ -0,0 +1,70 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pesq as pesq_backend
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def pesq(preds: Tensor, target: Tensor, fs: int, mode: str, keep_same_device: bool = False) -> Tensor:
+    r"""PESQ (Perceptual Evaluation of Speech Quality)
+
+    This is a wrapper for the pesq package [1].
+
+    Args:
+        preds:
+            shape ``[...,time]``
+        target:
+            shape ``[...,time]``
+        fs:
+            sampling frequency, should be 16000 or 8000 (Hz)
+        mode:
+            'wb' (wide-band) or 'nb' (narrow-band)
+        keep_same_device:
+            whether to move the pesq value to the device of preds
+
+    Returns:
+        pesq value of shape [...]
+
+    Example:
+        >>> from torchmetrics.functional.audio import pesq
+        >>> import torch
+        >>> preds = torch.randn(8000)
+        >>> target = torch.randn(8000)
+        >>> nb_pesq_val = pesq(preds, target, 8000, 'nb')
+        >>> wb_pesq_val = pesq(preds, target, 16000, 'wb')
+
+    References:
+        [1] https://github.com/ludlows/python-pesq
+    """
+    _check_same_shape(preds, target)
```

Comment:
should we also check they are 1D or 2D?
```suggestion
    _check_same_shape(preds, target)
    if preds.ndim > 2:
         raise RunTimeError('Got input with {preds.ndim} dimension, but expected it to either be 1D or 2D')
```

Reply:
All is ok. Even greater than 2D.

Reply:
if the shape is something like [x,y,z], the last dim is regarded as the time dim.
so a result of shape [x,y] will be returned